arbnode/execution/sequencer.go

```diff
@@ -405,31 +410,41 @@ func (s *Sequencer) PublishTransaction(parentCtx context.Context, tx *types.Tran
 		return types.ErrTxTypeNotSupported
 	}
 
-	ctx, cancelFunc := s.ctxWithQueueTimeout(parentCtx)
+	queueTimeout := s.config().QueueTimeout
+	queueCtx, cancelFunc := ctxWithTimeout(parentCtx, queueTimeout)
 	defer cancelFunc()
+	// Just to be safe, make sure we don't run over twice the queue timeout
+	submissionStart := time.Now()
+	abortDeadline := submissionStart.Add(queueTimeout * 2)
+	abortCtx, cancel := context.WithDeadline(parentCtx, abortDeadline)
```

Comment:
Do we consider the case when queueTimeout is zero? In that scenario queueCtx effectively has no timeout whereas abortCtx times out instantly, this might be an unintended consequence. Maybe we can reuse `ctxWithTimeout` function to create abortCtx as well

Reply:
Oh good point. I wanted to avoid using ctxWithTimeout because the deadline is a bit more exact, but it's fine if the context is a bit too late to expire, which is what would happen if I used ctxWithTimeout. I'll change to that.

Reply:
I went through a couple iterations on how to correctly check if the abort deadline got hit, but I have what I think is a clean approach. Please take a look :)

Reply:
cool